### PR TITLE
Upgrade Synapse to 1.19.1

### PR DESCRIPTION
**Pull request by ulope**
## Description

This upgrades the Synapse requirement to 1.18.0 and fixes all backwards incompatibilities that have been introduced.
This is mainly related to the [alias semantics change (MSC2432)](matrix-org/matrix-doc#2432).

Fixes: raiden-network/raiden-service-bundle#180
Fixes: #6424

Due to this being backwards incompatible server side we need to come up with a migration strategy before merging this: #6443 [done]

### CODE REVIEW

File: raiden/network/transport/matrix/client.py [hunk 1]
```diff
@@ -81,38 +82,39 @@ def _rmmembers(self, user_id: str) -> None:
...
-        Unfortunately due to a limitation in the Matrix API it's not possible to query for all
-        aliases of a room. Only aliases for a specific server can be fetched, see:
-        https://github.com/matrix-org/synapse/issues/6908
+        With MSC2432[1] implemented only ``m.room.canonical_alias`` events exist.
+        They represent the server local canonical_alias.
 
-        Since in Raiden we always have server local aliases set, this method is sufficient for our
-        use case.
+        Since in Raiden broadcast rooms always have server a local alias set, this method is
+        sufficient for our use case.
```
Comment (karlb) [inline raiden/network/transport/matrix/client.py:97]: Something is wrong or hard to read in first part of this sentence.

File: raiden/tests/utils/transport.py [hunk 2]
```diff
@@ -145,67 +139,76 @@ def __getattr__(self, item):
...
+                raise AssertionError(
+                    f"Could not read credentials file '{credentials_file}': {ex}"
+                ) from ex
+        elif "admin_credentials" in config:
+            self.credentials = config["admin_credentials"]
+        else:
+            raise AssertionError(
+                "Either 'credentials_file' or 'admin_credentials' must be specified in "
+                "auth provider config."
+            )
```
Comment (karlb) [inline raiden/tests/utils/transport.py:161]: Do we really need all these cases, now?

File: raiden/tests/utils/transport.py [hunk 3]
```diff
@@ -145,67 +139,76 @@ def __getattr__(self, item):
...
-    def __init__(self, config, account_handler):
+    def __init__(self, config, account_handler) -> None:  # type: ignore
         self.account_handler = account_handler
         self.config = config
         self.hs_hostname = self.account_handler._hs.hostname
         self.log = logging.getLogger(__name__)
 
-    @defer.inlineCallbacks
-    def check_password(self, user_id, password):
+    async def check_password(self, user_id: str, password: str) -> bool:
```
Comment (karlb) [inline raiden/tests/utils/transport.py:200]: Did synapse change away from twisted, or does twisted work with the async syntax, now?

File: raiden/utils/cli.py [hunk 4]
```diff
@@ -423,13 +425,27 @@ def apply_config_file(
...
+                ...
+            ],
+            "all_servers": [
+                "url1",
+                "url2",
+                ...
+            ]
+        }
+
+    Which of the two lists is returned is controlled by the ``server_list_type`` argument.
```
Comment (karlb) [inline raiden/utils/cli.py:448]: We could also just return all and let the caller choose by indexing the returned dict.

**Review by karlb — APPROVED** [review]
Looks good, I just left some random comments as usual.

File: raiden/tests/utils/transport.py (hunk 2 above)
Comment (ulope) [reply raiden/tests/utils/transport.py:161]: This now covers both cases, the one used in the tests as well as the RSB setup.
The plan is to have this code only once and not two slightly different versions in here and in the RSB.

File: raiden/tests/utils/transport.py (hunk 3 above)
Comment (ulope) [reply raiden/tests/utils/transport.py:200]: They are slowly converting to asyncio. 
This API has already been changed.

I'm not sure what they're using as a compatibility layer, but one of the twisted maintainers works for them so I assume they have that covered.

**Comment by ulope** [comment]
BTW the scenario player test can't succeed until the RSB PR is merged, which needs this PR merged.
I just love circular dependencies...

File: raiden/utils/cli.py (hunk 4 above)
Comment (karlb) [reply raiden/utils/cli.py:448]: It's not. I just like to thing aloud when reviewing PRs.

**Pull request merged**